pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -196,6 +196,20 @@ func (w *Workflow) GetExecSpec() *WorkflowSpec {
 	return &w.Spec
 }
 
+func (w *Workflow) HasArtifactGC() bool {
+	artifactDefinedGC := false
+	for _, template := range w.Spec.Templates {
+		for _, artifact := range template.Outputs.Artifacts {
```

Comment:
GetOutputs()

Reply:
@alexec GetOutputs() is a method of NodeStatus, not Template. Did you want me to implement Template.GetOutputs()?

Reply:
Only if outputs can be nil